azurelinuxagent/ga/exthandlers.py

```diff
@@ -97,12 +102,14 @@ def get_traceback(e):
 
 def validate_has_key(obj, key, fullname):
     if key not in obj:
-        raise ExtensionError("Missing: {0}".format(fullname))
+        raise ExtensionStatusError("Invalid status format by extension: Missing {0} key".format(fullname),
```

Comment:
What is `fullname` here in relation to `key`? `key` is missing from the dictionary, shouldn't the error message be `"Missing {0} key".format(key)`?

Reply:
The full name is the full path of the key that you are searching for. The use case was better to error reporting.

It essentially means - "Missing {0} key from the {1} node".format(key, obj), but the obj node is currently a dictionary, and we needed a way to explain that. Thus, the fullname helps in clarifying whats missing from where (eg: formattedMessage/lang).

Eg: `validate_has_key(formatted_message, 'lang', 'formattedMessage/lang')`

Reply:
Done - renamed to `full_key_path`. To describe the path.